addons/mrp/views/mrp_bom_views.xml

```diff
@@ -283,6 +283,57 @@
             </field>
         </record>
 
+        <record id="mrp_bom_line_form_action" model="ir.actions.act_window">
+            <field name="name">Used in BoM</field>
+            <field name="res_model">mrp.bom.line</field>
+            <field name="path">used_in_boms</field>
+            <field name="domain">[]</field>
+            <field name="view_mode">list</field>
+        </record>
+
+        <record id="mrp_bom_line_tree_view" model="ir.ui.view">
+            <field name="name">mrp.bom.line.list</field>
+            <field name="model">mrp.bom.line</field>
+            <field name="arch" type="xml">
+                <list create="False" delete="False" js_class="bom_line_list">
+                    <field name="bom_id" column_invisible="True"/>
+                    <field name="bom_product_tmpl_id"/>
+                    <field name="bom_code" optional="show"/>
+                    <field name="bom_type"/>
+                    <field name="bom_product_id" groups="product.group_product_variant" optional="hide"/>
+                    <field name="bom_product_id" groups="!product.group_product_variant" column_invisible="1"/>
```

Comment:
Why do you need this field to be placed two times in the view ?

Reply:
Very well, I've removed the second line :)